### PR TITLE
Fix alt+space opening system menu and sending keys to terminal

### LINKED ISSUE #10935: alt+space displays system menu *and* sends key binding to Terminal
### Windows Terminal version (or Windows build number)

1.11

### Other Software

_No response_

### Steps to reproduce

Press alt-space!

### Expected Behavior

_No response_

### Actual Behavior

image

/cc @FWest98 possible fallout from #10799?

### PR DESCRIPTION
**Pull request by PankajBhojwani**
If both of the following are true

1. alt+space is not explicitly unbound
2. alt+space is not bound to a command

Then the window procedure will handle the alt+space to open up the context menu.
In this case, we need to make sure we don't send the keys to terminal.

Closes #10935

### CODE REVIEW

**Comment by lhecker** [comment]
Isn't this equal to having a default key binding for Alt+Space which is bound to a no-op action?

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 1]
```diff
@@ -903,6 +903,23 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             return;
         }
 
+        if (vkey == VK_SPACE && modifiers.IsAltPressed())
+        {
+            if (const auto bindings = _settings.KeyBindings())
+            {
+                if (!bindings.IsKeyChordExplicitlyUnbound({ modifiers.IsCtrlPressed(), modifiers.IsAltPressed(), modifiers.IsShiftPressed(), modifiers.IsWinPressed(), vkey, 0 }))
```
Comment (lhecker) [inline src/cascadia/TerminalControl/TermControl.cpp:910]: Similarly to `_TryHandleKeyBinding` you should pass the scanCode here if you can.

Comment (PankajBhojwani) [reply src/cascadia/TerminalControl/TermControl.cpp]: Ah yes, thanks for the catch!

**Comment by DHowett** [comment]
Here's a really silly thought. What if we just add a new "openSystemMenu" binding? We could be done with this whole issue and allow people to re-bind it to whatever they want! All we'd need to do is move `_OpenSystemMenu` to IslandWindow, hook up some events from AppHost/AppLogic/TerminalPage (ugh hopefully not all of them) and... we're off to the races! We can *also* do that work later nad mark this part with a big TODO that links to a workitem that says what I just said above^

**Comment by DHowett** [comment]
Perhaps there's a nod to https://github.com/microsoft/terminal/issues/5833 somewhere in there...

**Comment by zadjii-msft** [comment]
> What if we just add a new "openSystemMenu" binding? We could be done with this whole issue and allow people to re-bind it to whatever they want! 

that's a good idea

**Comment by PankajBhojwani** [comment]
> Here's a really silly thought. What if we just add a new "openSystemMenu" binding? We could be done with this whole issue and allow people to re-bind it to whatever they want! All we'd need to do is move _OpenSystemMenu to IslandWindow, hook up some events from AppHost/AppLogic/TerminalPage (ugh hopefully not all of them) and... we're off to the races! We can also do that work later and mark this part with a big TODO that links to a workitem that says what I just said above^

Sounds good to me! Added a work item for it for 1.12

**Review by lhecker — APPROVED** [review]

**Review by DHowett — COMMENTED** [review]
I am okay with this solution -- it's targeted and specific, but it does rely on explicit unbinding. I don't _love_ that, but it's better than the alternative.

Piping input directly into the terminal using one of the direct input listeners is always going to be fraught with peril.

@PankajBhojwani, alt+space works for the menu when you do this? Excellent.

**Review by zadjii-msft — APPROVED** [review]
Okay yea lets do this for now, but `openSystemMenu` is definitely the better solution here long-term

**Pull request merged**